server/src/domain/media/media.service.ts

```diff
@@ -98,21 +98,24 @@ export class MediaService {
     if (!asset) {
       return false;
     }
-    const resizePath = this.ensureThumbnailPath(asset, 'jpeg');
-    const webpPath = this.ensureThumbnailPath(asset, 'webp');
-    const encodedVideoPath = this.ensureEncodedVideoPath(asset, 'mp4');
+    const resizePath = this.ensureThumbnailPath(asset, 'jpeg', false);
+    const webpPath = this.ensureThumbnailPath(asset, 'webp', false);
+    const encodedVideoPath = this.ensureEncodedVideoPath(asset, 'mp4', false);
 
     if (asset.resizePath && asset.resizePath !== resizePath) {
```

Comment:
Maybe we should leave ensure path like you had it before, but then only call it if the asset has the file to begin with, etc. So move the ensure path calls inside of the if statements. And have one for if the asset has the path and a second nested is for if the path is different.